### PR TITLE
Update matrix submodule and add tests

### PR DESCRIPTION
**Pull request by jlecoeur**
(starting to pull small pieces from #13351)

This PR updates matrix submodule with pseudoinverse implementation (https://github.com/PX4/Matrix/pull/102) and adds tests and benchmarks.

### CODE REVIEW

File: src/systemcmds/tests/test_matrix.cpp [hunk 1]
```diff
@@ -757,3 +759,98 @@ bool MatrixTest::dcmRenormTests()
...
+	// Mock-up effectiveness matrix
+	const float B_quad_w[6][16] = {
+		{-0.5717536f,  0.43756646f,  0.5717536f, -0.43756646f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f},
+		{ 0.35355328f, -0.35355328f,  0.35355328f, -0.35355328f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f},
+		{ 0.28323701f,  0.28323701f, -0.28323701f, -0.28323701f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f},
+		{ 0.f,  0.f,  0.f,  0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f},
+		{ 0.f,  0.f,  0.f,  0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f},
+		{-0.25f, -0.25f, -0.25f, -0.25f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f, 0.f}
+	};
+	Matrix<float, 6, 16> B = Matrix<float, 6, 16>(B_quad_w);
```
Comment (jkflying) [inline src/systemcmds/tests/test_matrix.cpp]:
```suggestion
	Matrix<float, 6, 16> B(B_quad_w);
```

File: src/systemcmds/tests/test_matrix.cpp [hunk 2]
```diff
@@ -757,3 +759,98 @@ bool MatrixTest::dcmRenormTests()
...
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f},
+		{ 0.0f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f}
+	};
+	Matrix<float, 16, 6> A_check = Matrix<float, 16, 6>(A_quad_w);
```
Comment (jkflying) [inline src/systemcmds/tests/test_matrix.cpp]:
```suggestion
	Matrix<float, 16, 6> A_check(A_quad_w);
```

File: boards/px4/fmu-v2/default.cmake [hunk 3]
```diff
@@ -83,7 +83,7 @@ px4_add_board(
 		battery_status
 		sensors
 		vmount
-		vtol_att_control
+		#vtol_att_control
```
Comment (jkflying) [inline boards/px4/fmu-v2/default.cmake]: I don't see why this is necessary, similarly with the other V2 changes. You haven't touched any code here which would increase binary sizes for these platforms.

File: boards/holybro/durandal-v1/stackcheck.cmake [hunk 4]
```diff
@@ -99,21 +99,21 @@ px4_add_board(
 		reflect
 		sd_bench
 		shutdown
-		tests # tests and test runner
+		# tests # tests and test runner
```
[the remark this answers is not in the document]
Comment (dagar) [reply boards/holybro/durandal-v1/stackcheck.cmake]: Things get much bigger in the stackcheck build. https://nuttx.org/doku.php?id=wiki:howtos:run-time-stack-checking

File: boards/px4/fmu-v2/default.cmake (hunk 3 above)
Comment (dagar) [inline boards/px4/fmu-v2/default.cmake]:
```suggestion
		vtol_att_control
```

Comment (dagar) [reply boards/px4/fmu-v2/default.cmake]: Reverting this to see what bloaty says about fmu-v2.

Comment (jlecoeur) [reply boards/px4/fmu-v2/default.cmake]: The v2 build was failing so I cherry-picked this commit from #13351. It is possible that I removed more than is necessary for this PR alone

Comment (jlecoeur) [reply boards/px4/fmu-v2/default.cmake]: Looks like we can keep it in this PR :+1

File: boards/holybro/durandal-v1/stackcheck.cmake (hunk 4 above)
[the remark this answers is not in the document]
Comment (jlecoeur) [reply boards/holybro/durandal-v1/stackcheck.cmake]: I removed a few matrix benchmarks that were running geninv on varying matrix sizes and using a lot of flash

**Comment by jlecoeur** [comment]
Pretty decent performance boost with the durandal :+1: 

holybro_durandal-v1_default:
```
matrix 6x16 pseudo inverse (all non-zero columns): 1000 events, 94156us elapsed, 94.16us avg, min 93us max 98us 0.835us rms
matrix 6x16 pseudo inverse (4 non-zero columns): 1000 events, 55353us elapsed, 55.35us avg, min 54us max 74us 2.104us rms
```

px4_fmu-v5x_default:
```
matrix 6x16 pseudo inverse (all non-zero columns): 1000 events, 209213us elapsed, 209.21us avg, min 207us max 211us 0.937us rms
matrix 6x16 pseudo inverse (4 non-zero columns): 1000 events, 133568us elapsed, 133.57us avg, min 120us max 159us 16.082us rms
```

Things get much slower with stackcheck enabled:
holybro_durandal-v1_stackcheck:
```
matrix 6x16 pseudo inverse (all non-zero columns): 1000 events, 416037us elapsed, 416.04us avg, min 416us max 417us 0.188us rms
matrix 6x16 pseudo inverse (4 non-zero columns): 1000 events, 260581us elapsed, 260.58us avg, min 245us max 314us 27.937us rms
```

**Pull request merged**